src/Tasks/Microsoft.NET.Build.Tasks/PackageInfo.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Xml;
+using System.Globalization;
+using System.Xml.Linq;
+
+namespace Microsoft.NET.Build.Tasks
+{
+    internal class PackageInfo : IEquatable<PackageInfo>
+    {
+        internal string Name { get; }
+
+        internal string Version { get; }
+
+        internal PackageInfo(string _Name, string _Version)
+        {
+            Name = _Name;
+            Version = _Version;
+        }
+
+        public override int GetHashCode()
+        {
+            return Name.GetHashCode() | Version.GetHashCode();
+        }
+
+        public bool Equals(PackageInfo pkg)
+        {
+            return Name.Equals(pkg.Name) && Version.Equals(pkg.Version);
```

Comment:
I think you want case-insensitive comparison here for Name.  NuGet package ids are case-insensitive.